src/System.Text.RegularExpressions/tests/RegexParserTests.cs

```diff
@@ -0,0 +1,1029 @@
+        [InlineData(@"@""\b\d{2,}\b\D+""", RegexOptions.None, null)]
+        [InlineData(@"@""(00\s){2,4}""", RegexOptions.None, null)]
+        [InlineData(@"@""\b\w*?oo\w*?\b""", RegexOptions.None, null)]
+        [InlineData(@"@""\b\w+?\b""", RegexOptions.None, null)]
+        [InlineData(@"@""^\s*(System.)??Console.Write(Line)??\(??""", RegexOptions.None, null)]
+        [InlineData(@"@""(System.)??""", RegexOptions.None, null)]
+        [InlineData(@"@""\b(\w{3,}?\.){2}?\w{3,}?\b""", RegexOptions.None, null)]
+        [InlineData(@"@""\b[A-Z](\w*?\s*?){1,10}[.!?]""", RegexOptions.None, null)]
+        [InlineData(@"@""b.*([0-9]{4})\b""", RegexOptions.None, null)]
+        [InlineData(@"@""\b.*?([0-9]{4})\b""", RegexOptions.None, null)]
+        [InlineData(@"@""(a?)*""", RegexOptions.None, null)]
+        [InlineData(@"@""(a\1|(?(1)\1)){0,2}""", RegexOptions.None, null)]
+        [InlineData(@"@""(a\1|(?(1)\1)){2}""", RegexOptions.None, null)]
+        [InlineData(@"@""(\w)\1""", RegexOptions.None, null)]
+        [InlineData(@"@""(?<char>\w)\k<char>""", RegexOptions.None, null)]
+        [InlineData(@"@""(?<2>\w)\k<2>""", RegexOptions.None, null)]
+        [InlineData(@"@""(?<1>a)(?<1>\1b)*""", RegexOptions.None, null)]
+        [InlineData(@"@""\b(\p{Lu}{2})(\d{2})?(\p{Lu}{2})\b""", RegexOptions.None, null)]
+        [InlineData(@"@""\bgr[ae]y\b""", RegexOptions.None, null)]
+        [InlineData(@"@""\b((?# case sensitive comparison)D\w+)\s(?ixn)((?#case insensitive comparison)d\w+)\b""", RegexOptions.None, null)]
+        [InlineData(@"@""\{\d+(,-*\d+)*(\:\w{1,4}?)*\}(?x) # Looks for a composite format item.""", RegexOptions.None, null)]
+        // Negative tests
+        [InlineData(@"@""cat([a-\d]*)dog""", RegexOptions.None, RegexParseError.BadClassInCharRange)]
+        [InlineData(@"@""\k<1""", RegexOptions.None, RegexParseError.UnrecognizedEscape)]
+        [InlineData(@"@""\k<""", RegexOptions.None, RegexParseError.UnrecognizedEscape)]
+        [InlineData(@"@""\k""", RegexOptions.None, RegexParseError.MalformedNameRef)]
+        [InlineData(@"@""\1""", RegexOptions.None, RegexParseError.UndefinedBackref)]
+        [InlineData(@"@""(?')""", RegexOptions.None, RegexParseError.InvalidGroupName)]
+        [InlineData(@"@""(?<)""", RegexOptions.None, RegexParseError.InvalidGroupName)]
+        [InlineData(@"@""(?)""", RegexOptions.None, RegexParseError.QuantifyAfterNothing)]
+        [InlineData(@"@""(?>""", RegexOptions.None, RegexParseError.NotEnoughParens)]
+        [InlineData(@"@""(?<!""", RegexOptions.None, RegexParseError.NotEnoughParens)]
+        [InlineData(@"@""(?<=""", RegexOptions.None, RegexParseError.NotEnoughParens)]
+        [InlineData(@"@""(?!""", RegexOptions.None, RegexParseError.NotEnoughParens)]
+        [InlineData(@"@""(?=""", RegexOptions.None, RegexParseError.NotEnoughParens)]
+        [InlineData(@"@""(?imn )""", RegexOptions.None, RegexParseError.UnrecognizedGro
```

Comment:
this should be no problem for you guys, i don't tink.  You don't use actual stack recursion when parsing.  This was a special test for Roslyn because my parser is recursive descent, and so i needed to make sure something complex like this bailed out properly without causing a StackOverflowException.

Reply:
You are talking about the DeepRecursion test, right?

Reply:
Why not keep it, tests ideally shouldn't assume the choice of implementation